ZenPacks/zenoss/ZenPackLib/lib/wrapper/ComponentPathReporter.py

```diff
@@ -9,23 +9,59 @@
 from zope.component import adapts
 from zope.interface import implements
 from Products.Zuul.catalog.interfaces import IPathReporter
-from Products.Zuul.catalog.paths import DefaultPathReporter, relPath
-from ..base.ComponentBase import ComponentBase
-
+from Products.Zuul.catalog.paths import (
+     DefaultPathReporter,
+     DevicePathReporter,
+     ServicePathReporter,
+     InterfacePathReporter,
+     ProcessPathReporter,
+     ProductPathReporter,
+     relPath
+     )
 
-class ComponentPathReporter(DefaultPathReporter):
+from ..base.ComponentBase import ComponentBase
+from ..base.Component import (
+     HWComponent,
+     IpInterface,
+     OSProcess,
+     Service,
+)
 
-    """Global catalog path reporter adapter factory for components."""
 
+class ComponentPathMixIn(DefaultPathReporter):
+    """Global catalog path reporter override"""
     implements(IPathReporter)
-    adapts(ComponentBase)
 
     def getPaths(self):
-        paths = super(ComponentPathReporter, self).getPaths()
+        paths = super(ComponentPathMixIn, self).getPaths()
 
         for facet in self.context.get_facets():
             rp = relPath(facet, facet.containing_relname)
             paths.extend(rp)
 
         return paths
 
+
+class ComponentPathReporter(ComponentPathMixIn):
+    """Global catalog path reporter adapter factory for basic components."""
+    adapts(ComponentBase)
+
+
+class ProductPathReporter(ProductPathReporter, ComponentPathMixIn):
```

Comment:
You shouldn't redefine ProductPathReporter like this. It changes the meaning of what ProductPathReporter is from the beginning to the end of the module. It would be better to import ProductPathReporter as ProductPathReporterBase to eliminate that potential confusion.

I don't think this adapter is going to work as expected. When ProductPathReporter.getPaths() is called, it will end up calling ProductPathReporterBase.getPaths(), not ComponentPathMixin.getPaths() because ProductPathReporterBase is the first superclass, and it has a getPaths method.

These same issues also exist for InterfacePathReported, ProcessPathReported, and ServicePathReporter below.